casper/src/main/scala/coop/rchain/casper/CasperSnapshot.scala

```diff
@@ -1,51 +1,29 @@
 package coop.rchain.casper
 
-import coop.rchain.blockstorage.dag.DagRepresentation
-import coop.rchain.casper.protocol._
-import coop.rchain.crypto.signatures.Signed
+import com.google.protobuf.ByteString
 import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.models.BlockMetadata
 import coop.rchain.models.Validator.Validator
+import coop.rchain.rspace.hashing.Blake2b256Hash
 
 /**
-  * Casper snapshot is a state that is changing in discrete manner with each new block added.
-  * This class represents full information about the state. It is required for creating new blocks
-  * as well as for validating blocks.
+  * [[ParentsMergedState]] represents required data to create or validate a block from the view of
+  * parent relations (justifications) or for the new block, latest messages which will be used as justifications.
+  *
+  * @param justifications block justifications (latest blocks for the new block)
+  * @param fringe finalized fringe seen (finalized) by parents
+  * @param fringeState finalized fringe (merged) state
+  * @param bondsMap bonds map of validators on fringe state
+  * @param rejectedDeploys rejected deploys from blocks finalized with [[fringe]] blocks
+  * @param maxBlockNum maximum block height from parent blocks
+  * @param maxSeqNums latest sequence numbers for bonded validators
   */
-final case class CasperSnapshot(
-    dag: DagRepresentation,
-    lastFinalizedBlock: BlockHash,
-    lca: BlockHash,
-    tips: IndexedSeq[BlockHash],
-    parents: List[BlockMessage],
-    justifications: Set[Justification],
-    deploysInScope: Set[Signed[DeployData]],
-    maxBlockNum: Long,
-    maxSeqNums: Map[Validator, Int],
-    onChainState: OnChainCasperState
-)
-
-final case class OnChainCasperState(
-    shardConf: CasperShardConf,
+final case class ParentsMergedState(
```

Comment:
Rename this file to `ParentsMergedState.scala`?

Reply:
I'm sustaining of file renaming/moving to reduce possible conflicts. I'll create a separate PR for reorganizing files in casper project.
Also I want to merge this PR first and leave your PR #3720 to resolve conflicts.